File: solutions/506.cpp
class Solution {
public:
    vector<string> findRelativeRanks(vector<int>& score) {
        vector<pair<int, int>> v;
        int n = score.size();
        for (int i = 0; i < n; ++i) {
            v.push_back(make_pair(score[i], i));
        }
        sort(v.begin(), v.end(), greater<pair<int, int>>());
        
        vector<string> res(n, "");
        for (int i = 0; i < n; ++i) {
            int index = v[i].second;
            if (i == 0) res[index] = "Gold Medal";
            else if (i == 1) res[index] = "Silver Medal";
            else if (i == 2) res[index] = "Bronze Medal";
            else res[index] = to_string(i + 1);
        }
        return res;
    }
};

File: rust_solutions/506.rs
impl Solution {
    pub fn find_relative_ranks(mut score: Vec<i32>) -> Vec<String> {
        let mut d = std::collections::HashMap::new();
        for i in 0..score.len() {
            d.insert(score[i], i);
        }
        score.sort();
        score.reverse();
        let mut ret = vec!["".to_string(); score.len()];
        for i in 0..score.len() {
            let rank = match i {
                0 => "Gold Medal".to_owned(),
                1 => "Silver Medal".to_owned(),
                2 => "Bronze Medal".to_owned(),
                _ => (i + 1).to_string(),
            };
            ret[*d.get(&score[i]).unwrap()] = rank;
        }
        ret
    }
}
